pkl-core/src/main/java/org/pkl/core/ast/expression/generator/GeneratorEntryNode.java

```diff
@@ -108,15 +107,6 @@ private void addListingEntry(VirtualFrame frame, ObjectData data, int parentLeng
           .build();
     }
 
-    doAdd(index, data);
-  }
-
-  private void doAdd(Object key, ObjectData data) {
-    if (EconomicMaps.put(data.members, key, member) != null) {
-      CompilerDirectives.transferToInterpreter();
-      throw duplicateDefinition(key, member);
-    }
```

Comment:
What's the rationale behind getting rid of this check?

Reply:
It's not removed, just moved from node classes to class `ObjectData` (see method `addMember`).